src/iso19111/metadata.cpp

```diff
@@ -1249,11 +1257,34 @@ static const struct utf8_to_lower *get_ascii_replacement(const char *c_str) {
 // ---------------------------------------------------------------------------
 
 //! @cond Doxygen_Suppress
-std::string Identifier::canonicalizeName(const std::string &str) {
+
+static bool matchesLowerCase(const char *c_str, const char *needle) {
+    size_t i = 0;
+    for (; c_str[i] && needle[i]; ++i) {
+        if (lower(c_str[i]) != lower(needle[i])) {
+            return false;
+        }
+    }
+    return needle[i] == 0;
```

Comment:
`matchesLowerCase("Java", "javaScript");` returns `false`, but `matchesLowerCase("JavaScript", "java");` returns `true`. Is that intentional?
Otherwise
```suggestion
    return c_str[i] == 0 && needle[i] == 0;
```

Reply:
> Is that intentional?

yes, it checks if it finds needle (substring) within the string. Code comment added